README.md

```diff
@@ -334,39 +241,41 @@ apt::sources:
 
 It is recommended to read the manpage 'apt_preferences(5)'
 
-### Testing
-
-The apt module is mostly a collection of defined resource types, which provide reusable logic for managing Apt. It provides smoke tests for testing functionality on a target system, as well as spec tests for checking a compiled catalog against an expected set of resources.
-
-#### Example Test
-
-This test sets up a Puppet Labs Apt repository. Start by creating a new smoke test, called puppetlabs-apt.pp, in the apt module's test folder. In this test, declare a single resource representing the Puppet Labs Apt source and GPG key:
-
-```
-apt::source { 'puppetlabs':
-  location   => 'http://apt.puppetlabs.com',
-  repos      => 'main',
-  key        => '1054B7A24BD6EC30',
-  key_server => 'pgp.mit.edu',
-}
-```    
-
-This resource creates an Apt source named puppetlabs and gives Puppet information about the repository's location and the key used to sign its packages. Puppet leverages Facter to determine the appropriate release, but you can set this directly by adding the release type.
+####apt::ppa
 
-Check your smoke test for syntax errors:
+* `ensure`: Whether to add or remove the PPA. Valid values are 'present' or 'absent'. Defaults to 'present'.
+* `options`: Options to pass to `add-apt-repository`. OS-dependent defaults are defined in `apt::params`.
+* `release`: OS-release, used in the filename of the generated sources.list.d file. Defaults to `$::lsbdistcodename`.
+* `package_name`: The package to install `add-apt-repository`. OS-dependent defaults are defined in `apt::params`.
+* `package_manage`: Whether or not to manage the package for `add-apt-repository`. Defaults to false.
 
-`$ puppet parser validate tests/puppetlabs-apt.pp`
+####apt::setting
 
-If you receive no output from that command, it means nothing is wrong. Then, apply the code:
+* `priority`: Integer or zero-padded integer setting the file priority. Defaults to 50.
+* `ensure`: Whether to add or remove the file. Valid values are 'present', 'absent', and 'file'. Defaults to file.
+* `source`: The source for the file. Exactly one of `content` and `source` must be specified.
+* `content`: The content for the file. Exactly one of `content` and `source` must be specified.
+* `notify_update`: Boolean for whether or not this `apt::setting` should notify `Class['apt::update']`. Defaults to true.
```

Comment:
Perhaps lets say here "Boolean for whether or not this `apt::setting` should trigger an `apt-get update`. Defaults to true"